base/jobs/docker/1.0/py3/Dockerfile.cpu

```diff
@@ -1,125 +1,146 @@
 
-# Upgrading pip and creating symbolic link for python3
-RUN ${PIP} --no-cache-dir install --upgrade pip setuptools
-RUN ln -s $(which ${PYTHON}) /usr/local/bin/python \
- && ln -s $(which ${PIP}) /usr/bin/pip
+LABEL maintainer="Amazon Braket" \
+      major_version="1"
+
+ENV DEBIAN_FRONTEND=noninteractive \
+    DEBCONF_NONINTERACTIVE_SEEN=true \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    PYTHONIOENCODING=UTF-8 \
+    LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8 \
+    LD_LIBRARY_PATH=/usr/local/openmpi/lib:/usr/local/lib:$LD_LIBRARY_PATH \
+    PATH=/usr/local/openmpi/bin/:$PATH \
+    SAGEMAKER_PROGRAM=braket_container.py
+
+# Copy from previous stages
+COPY --from=openmpi-builder /usr/local /usr/local
+COPY --from=python-builder /usr/local /usr/local
+
+# Install build deps
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    openssh-client \
+    openssh-server \
+    build-essential \
+    ca-certificates \
+    curl \
+    git \
+    libtemplate-perl \
+    libssl3 \
+    openssl \
+    unzip \
+    libgtk2.0-dev \
+    cmake \
+    protobuf-compiler \
+    && rm -rf /var/lib/apt/lists/* \
+              /usr/share/doc \
+              /usr/share/man \
+              /usr/share/locale \
+              /var/cache/debconf \
+              /var/lib/dpkg/info \
+              /var/log/* \
+    && rm -rf /usr/local/lib/*.a \
+    && sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd \
+    && mkdir -p /root/.ssh/ /var/run/sshd \
+    && ssh-keygen -q -t rsa -N '' -f /root/.ssh/id_rsa \
+    && cp /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys \
+    && printf "Host *\n  StrictHostKeyChecking no\n" >> /root/.ssh/config \
+    && ln -sf $(which python3) /usr/local/bin/python \
+    && ln -sf $(which pip3) /usr/bin/pip
 
 WORKDIR /
 
-RUN apt-get update && apt-get -y install cmake protobuf-compiler
-
-# Copy the dependencies file
+# Install Python dependencies
 COPY requirements.txt /
-# Installing our custom python libraries
-RUN ${PIP} install --no-cache --upgrade \
-        -r requirements.txt
-
-RUN HOME_DIR=/root \
- && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
- && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
- && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
- && chmod +x /usr/local/bin/testOSSCompliance \
- && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
- && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
- && rm -rf ${HOME_DIR}/oss_compliance*
-
-# Setup our entry point
-
+RUN pip3 install --no-cache-dir -r requirements.txt \
+    && rm -rf /root/.cache/pip \
+    && find /usr/local/lib/python* -type d -name "__pycache__" -exec rm -rf {} + 2>/dev/null || true
+
+# Run compliance script
+RUN curl -o /root/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
```

Comment:
nit: LGTM, let's add the ARG HOME_DIR=/root above since its used in multiple places.